core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -943,4 +955,71 @@ public String toString() {
       return subchannel.getLogId().toString();
     }
   }
+
+  @VisibleForTesting
+  static final class ManagedChannelPhantom extends PhantomReference<ManagedChannelImpl> {
```

Comment:
WeakRefs are cleared on Minor GC compaction, even if the Channel is still around.  Phantoms are only cleared when their referent is GC'd.  Technically there is no need to clean a Phantom, since they don't keep their referent alive, and `get()` always returns null.